app/controllers/posts_controller.rb

```diff
@@ -1,30 +1,54 @@
 class PostsController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :set_post, only: :show
+
   def index
     @user = User.find(params[:user_id])
+    if current_user.nil?
+      redirect_to new_user_session_path
+      return
+    end
+    @posts = if current_user.role?
+               @user.posts.includes(:comments)
+             else
+               @user.posts.includes(:comments, :author)
+             end
   end
 
   def show
+    @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
 
   def new
-    @id = current_user.id
     @post = Post.new
   end
 
   def create
-    @post = Post.new(title: post_params[:title], text: post_params[:text], user: current_user)
-    flash[:notice] = if post.save
-                       'Post created successfully'
-                     else
-                       'Error'
-                     end
-    redirect_to user_posts_path
+    @post = Post.new(post_params)
+    if @post.save
+      flash[:success] = 'Post Created Successfully'
+      redirect_to user_posts_url(@post)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] The post creation does not work correctly, and since it is impossible to create a post successfully, it is difficult to test if the `destroy` functionality for comments which the previous reviewer suggested is working correctly. 
Since comments can not exist without a post, kindly fix the issue with the post so that the next reviewer can check if only authorized users can delete comments 🙏